### PR TITLE
wg_engine: support resizing

### PR DESCRIPTION
**Pull request by SergeyLebedkin**
Added abillity to resize target (surface)
In case of resizing only render targets handles recreated 
Must be tested in web environment
Issue: https://github.com/thorvg/thorvg/issues/2985

### CODE REVIEW

**Comment by hermet** [comment]
@SergeyLebedkin can we separate the commit into two parts?

1. Resizing support enhancement
2. Revision(refactor) of device creation policy

**Comment by SergeyLebedkin** [comment]
> @SergeyLebedkin can we separate the commit into two parts?
> 
> 1. Resizing support enhancement
> 2. Revision(refactor) of device creation policy

Done

**Review by tinyjin — APPROVED** [review]
https://github.com/user-attachments/assets/c3308b56-fb8c-47e6-9703-c1de17a26066

Resize works again 🎉

File: src/renderer/wg_engine/tvgWgRenderer.h [hunk 1]
```diff
@@ -66,6 +66,7 @@ class WgRenderer : public RenderMethod
     void disposeObjects();
     void releaseSurfaceTexture();
 
+    void updateTargets(void* target, uint32_t width, uint32_t height, int type);
```
Comment (hermet) [inline src/renderer/wg_engine/tvgWgRenderer.h:69]: looks not use?

Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgRenderer.h]: fixed

**Review by hermet — APPROVED** [review]

**Pull request merged**